Chapter9/Chapter9/Controller/RestaurantTableViewController.swift

```diff
@@ -28,27 +30,65 @@ class RestaurantTableViewController: UIViewController {
     private let tickImageName = "tick"
     private let undoImageName = "undo"
     private let heartImageName = "heart-tick"
-    private var restaurants = [Restaurant]()
+    private var restaurantsMO = [RestaurantMO]()
+
     var activityController: UIActivityViewController?
+    var fetchResultController: NSFetchedResultsController<RestaurantMO>!
+    var searchController: UISearchController!
+    var searchResults: [Restaurant] = []
 
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        let restaurantGroup = RestaurantGroup()
-        restaurants = restaurantGroup.restaurants
+        RestaurantGroup.populateDb()
```

Comment:
If you put on this method, we will be called every time this screen will be initialised.